File: implementation/scipamato/public/web/src/main/java/ch/difty/scipamato/publ/web/resources/IcoMoonIconType.java
package ch.difty.scipamato.publ.web.resources;

import de.agilecoders.wicket.core.markup.html.bootstrap.image.IconType;
import org.jetbrains.annotations.NotNull;

public class IcoMoonIconType extends IconType {
    private static final long serialVersionUID = 1L;

    public static final IcoMoonIconType arrow_right = new IcoMoonIconType("arrow-right");
    public static final IcoMoonIconType link        = new IcoMoonIconType("link");

    private IcoMoonIconType(@NotNull String cssClassName) {
        super(cssClassName);
    }

    @NotNull
    public String cssClassName() {
        return "icon-" + this.getCssClassName();
    }
}

File: public/public-web/src/main/kotlin/ch/difty/scipamato/publ/web/resources/IcoMoonIconType.kt
package ch.difty.scipamato.publ.web.resources

import de.agilecoders.wicket.core.markup.html.bootstrap.image.IconType

class IcoMoonIconType private constructor(cssClassName: String) : IconType(cssClassName) {

    override fun cssClassName(): String = "icon-$cssClassName"

    companion object {
        private const val serialVersionUID = 1L
        val arrow_right = IcoMoonIconType("arrow-right")
        val link = IcoMoonIconType("link")
    }
}
